lib/Index/Index.cpp

```diff
@@ -698,12 +698,27 @@ class IndexSwiftASTWalker : public SourceEntityWalker {
     }
   }
 
+  void handleLiteralInitRefs(Expr *E) {
```

Comment:
I assume this isn't being walked by SemaAnnotator because it's implicit? Feels odd to me that `visitDeclReference` takes a `ReferenceMetaData` which has `IsImplicit` and *some* expressions we walk regardless, but not others. Though that bit looks to have been added a bit later on. Perhaps we should walk `LiteralExpr` despite being implicit as well?

Any thoughts @benlangmuir?

Reply:
Seem reasonable to me, I don't recall if there's any reason for the current behaviour other than historical.

Reply:
Fair enough, I'd personally prefer doing this in `SemaAnnotator` then - IMO `IndexSwiftASTWalker` should be much simpler than it is right now, with all the actual walking done in `SemaAnnotator` instead.

Reply:
This function is now removed; the changes above made this work without special handling here.